proof/crefine/RISCV64/SyscallArgs_C.thy

```diff
@@ -126,11 +124,109 @@ lemma ccorres_pre_getWorkUnits:
+
+lemma ccorres_pre_getConsumedTime:
+  assumes cc: "\<And>rv. ccorres r xf (P rv) (P' rv) hs (f rv) c"
+  shows   "ccorres r xf
+                  (\<lambda>s. (\<forall>rv. ksConsumedTime s = rv \<longrightarrow> P rv s))
+                  {s. \<forall>rv. ksConsumed_' (globals s) = rv \<longrightarrow> s \<in> P' rv }
+                          hs (getConsumedTime >>= (\<lambda>rv. f rv)) c"
+  apply (rule ccorres_guard_imp)
+    apply (rule ccorres_symb_exec_l)
+       defer
+       apply wp[1]
+      apply (rule getConsumedTime_sp)
+     apply (clarsimp simp: empty_fail_def getConsumedTime_def simpler_gets_def)
+    apply assumption
+   apply clarsimp
+   defer
+   apply (rule ccorres_guard_imp)
+     apply (rule cc)
+    apply clarsimp
+   apply assumption
+  apply (clarsimp simp: rf_sr_ksConsumed)
+  done
+
+lemma updateTimestamp_ccorres:
+  "ccorres dc xfdc \<top> UNIV [] updateTimeStamp (Call updateTimestamp_'proc)"
+sorry (* FIXME RT: updateTimestamp_ccorres *)
+
+lemma rf_sr_ksCurSC:
+  "(s, s') \<in> rf_sr \<Longrightarrow> ksCurSC_' (globals s') = Ptr (ksCurSc s)"
+  by (clarsimp simp: rf_sr_def cstate_relation_def Let_def)
+
+lemma ccorres_pre_getCurSc:
+  assumes cc: "\<And>rv. ccorres r xf (P rv) (P' rv) hs (f rv) c"
+  shows   "ccorres r xf
+                  (\<lambda>s. (\<forall>rv. ksCurSc s = rv \<longrightarrow> P rv s))
+                  {s. \<forall>rv. ksCurSC_' (globals s) = Ptr rv \<longrightarrow> s \<in> P' rv }
+                          hs (getCurSc >>= (\<lambda>rv. f rv)) c"
+  apply (rule ccorres_guard_imp)
+    apply (rule ccorres_symb_exec_l)
+       defer
+       apply wp[1]
+      apply (rule getCurSc_sp)
+     apply (clarsimp simp: empty_fail_def getCurSc_def simpler_gets_def)
+    apply assumption
+   apply clarsimp
+   defer
+   apply (rule ccorres_guard_imp)
+     apply (rule cc)
+    apply clarsimp
+   apply assumption
+  apply (clarsimp simp: rf_sr_ksCurSC)
+  done
+
+lemma sc_active_ccorres:
+  "ccorres dc xfdc
+     \<top> \<lbrace>\<acute>sc = Ptr scPtr\<rbrace> []
+     (scActive scPtr) (Call sc_active_'proc)"
+sorry (* FIXME RT: sc_active_ccorres *)
+
+lemma refill_sufficient_ccorres:
+  "ccorres dc xfdc
+     \<top> (\<lbrace>\<acute>sc = Ptr scPtr\<rbrace> \<inter> \<lbrace>\<acute>usage = usage\<rbrace>) []
+     (refillSufficient scPtr usage) (Call refill_sufficient_'proc)"
+sorry (* FIXME RT: refill_sufficient_ccorres *)
+
+lemma isCurDomainExpired_ccorres:
+  "ccorres dc xfdc \<top> UNIV [] isCurDomainExpired (Call isCurDomainExpired_'proc)"
+sorry (* FIXME RT: isCurDomainExpired_ccorres *)
+
 lemma preemptionPoint_ccorres:
   "ccorres (cintr \<currency> dc) (liftxf errstate id (K ()) ret__unsigned_long_')
       invs' UNIV []
       preemptionPoint (Call preemptionPoint_'proc)"
-  apply (cinit simp: workUnitsLimit_def whenE_def)
+  apply (cinit simp: Kernel_Config.workUnitsLimit_def whenE_def) (* FIXME RT: correct use of defn here? *)
```

Comment:
I assume on the C side we're just seeing the plain number, and on the Haskell side we're seeing `Kernel_Config.workUnitsLimit_def`. I think in that case unfolding is legitimate here -- we don't have a nice mechanism yet to make the C side symbolic.